assets/js/util/i18n.js

```diff
@@ -19,22 +19,189 @@
+ *
+ * @since 1.7.0
+ *
+ * @param {number} number The large number to prepare.
+ * @return {number} The prepared number.
+ */
+export const prepareForReadableLargeNumber = ( number ) => {
+	if ( 1000000 <= number ) {
+		return Math.round( number / 100000 ) / 10;
+	}
+
+	if ( 10000 <= number ) {
+		return Math.round( number / 1000 );
+	}
+
+	if ( 1000 <= number ) {
+		return Math.round( number / 100 ) / 10;
+	}
+	return number;
+};
+
+/**
+ * Formats a large number for shortened display.
+ *
+ * @since 1.0.0
+ *
+ * @param {number} number The large number to format.
+ * @return {string} The formatted number.
+ */
+export const readableLargeNumber = ( number ) => {
+	const withSingleDecimal = {
+		minimumFractionDigits: 1,
+		maximumFractionDigits: 1,
+	};
+
+	// Numbers over 1,000,000 round normally and display a single decimal unless the decimal is 0.
+	if ( 1000000 <= number ) {
+		return sprintf(
+			// translators: %s: an abbreviated number in millions.
+			__( '%sM', 'google-site-kit' ),
+			numberFormat( prepareForReadableLargeNumber( number ), number % 10 === 0 ? {} : withSingleDecimal )
+		);
+	}
+
+	// Numbers between 10,000 and 1,000,000 round normally and have no decimals
+	if ( 10000 <= number ) {
+		return sprintf(
+			// translators: %s: an abbreviated number in thousands.
+			__( '%sK', 'google-site-kit' ),
+			numberFormat( prepareForReadableLargeNumber( number ) )
+		);
+	}
+
+	// Numbers between 1,000 and 10,000 round normally and display a single decimal unless the decimal is 0.
+	if ( 1000 <= number ) {
+		return sprintf(
+			// translators: %s: an abbreviated number in thousands.
+			__( '%sK', 'google-site-kit' ),
+			numberFormat( prepareForReadableLargeNumber( number ), number % 10 === 0 ? {} : withSingleDecimal )
+		);
+	}
+
+	return numberFormat( number );
+};
+
+/**
+ * Formats a number with unit using the JS Internationalization Number Format API.
+ *
+ * In addition to the supported 'style' values of the lower-level `numberFormat` function, this function
+ * supports two additional 'style' values 'metric' and 'seconds'.
+ *
+ * Another differentiation in behavior is that by default the function will use 'metric' formatting instead
+ * of 'decimal' formatting.
+ *
+ * @since n.e.x.t
+ *
+ * @param {number|string}                     number    The number to format.
+ * @param {(Intl.NumberFormatOptions|string)} [options] Formatting options or unit shorthand.
+ *                                                      Possible shorthand values are '%', 's',
+ *                                                      or a currency code.
+ * @return {string} The formatted number with unit.
+ */
+export const numFmt = ( number, options = {} ) => {
+	// Cast parsable values to numeric types.
+	number = isFinite( number ) ? number : Number( number );
+
+	if ( ! isFinite( number ) ) {
+		// eslint-disable-next-line no-console
+		console.warn( 'Invalid number', number, typeof number );
+		return null;
```

Comment:
Do we want to return `null` here? The return tag shows `string` as the only return type.

We should probably change `number` to `0` in this case and let the function continue to format it according to the options.